public/docs/ts/latest/guide/style-guide.jade

```diff
@@ -0,0 +1,672 @@
+
+  **Why?** Small functions are easier to maintain.
+
+  **Why?** Small functions help avoid hidden bugs that come with large functions that share variables with external scope, create unwanted closures, or unwanted coupling with dependencies.
+
+<a href="#toc">Back to top</a>
+
+:marked
+  ## Modules
+
+  ### Avoid Naming Collisions
+  <a id="020"></a>
+  #### Style 020
+
+    - Use unique naming conventions.
+
+    **Why?** Unique names help avoid module name collisions. We import and export modules by the file in which they are contained. Thus the name of the files and their folders is important.
+
+    **Why?** Names of folders and files should clear convey their intent. For example, `app/speakers/speaker-list.component.ts` may contain a component that manages a list of speakers. The folder and file names clearly convey the meaning, and thus the module import is clear as `import { SpeakerComponent } from './speakers/speaker-list.component'`.
+
+<a href="#toc">Back to top</a>
+
+:marked
+  ## Components
+
+  ### Member Sequence
+  <a id="033"></a>
+  #### Style 033
+
+    - Place properties up top, then public functions, then private functions, alphabetized, and not spread through the component code.
+
+      **Why?** Placing members in a consistent sequence makes it easy to read and helps you instantly identify which members of the component serve which purpose.
+
+    ```typescript
+    /* recommended */
+    export class ToastComponent implements OnInit {
+      // private fields
+      private _defaults = {
+        title: '',
+        message: 'May the Force be with You'
+      };
+      private _toastElement: any;
+
+      // public properties
+      title: string;
+      message: string;
+
+      // ctor
+      constructor(toastService: ToastService) {
+        toastService.activate = this.activate.bind(this);
+      }
+
+      // public functions
+      activate(message = this._defaults.message, title = this._defaults.title) {
+        this.title = title;
+        this.message = message;
+        this._show();
+      }
+
+      ngOnInit() {
+        this._toastElement = document.getElementById('toast');
+      }
+
+      // private functions
+      private _show() {
+        console.log(this.message);
+        this._toastElement.style.opacity = 1;
+        this._toastElement.style.zIndex = 9999;
+
+        window.setTimeout(() => this._hide(), 2500);
+      }
+
+      private _hide() {
+        this._toastElement.style.opacity = 0;
+        window.setTimeout(() => this._toastElement.style.zIndex = 0, 400);
+      }
+    }
+    ```
+
+<a href="#toc">Back to top</a>
+
+:marked
+  ### Defer Logic to Services
+  <a id="035"></a>
+  #### Style 035
+
+    - Defer logic in a component by delegating to services.
+
+    - Define a component for a view, and try not to reuse the component for other views. Instead, move reusable logic to factories and keep the component simple and focused on its view.
```

Comment:
this sounds weird to me. In my mind there are two kinds of components:

1/ routable components that are usually very application specific and are not meant to be reused even within the application.
2/ reusable components (application agnostic ones as buttons, or the whole material suite, and application-specific ones like profile/product widgets with pop-overs, etc).

the purpose of the service is to facilitate data delivary to these components